### PR TITLE
Mnt cleanup pylab setup

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: lib/matplotlib/backends/__init__.py [hunk 1]
```diff
@@ -7,29 +7,49 @@
 import inspect
 import warnings
 
-# ipython relies on interactive_bk being defined here
-from matplotlib.rcsetup import interactive_bk
 
-__all__ = ['backend','show','draw_if_interactive',
-           'new_figure_manager', 'backend_version']
```
Comment (WeatherGod) [inline lib/matplotlib/backends/__init__.py]: why get rid of this?

Comment (tacaswell) [inline lib/matplotlib/backends/__init__.py]: Most of these things are no longer exist in the module so it would raise an AttributeError.

**Comment by tacaswell** [comment]
I left pretty through commit messages on all of these changes.

**Comment by WeatherGod** [comment]
Ah, indeed. Ok, this looks fine to me. I take it none of this is supposed to be backported?

**Pull request merged**